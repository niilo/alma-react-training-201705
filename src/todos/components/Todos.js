import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import InputBar from '../../components/InputBar.js'
import Todo from './Todo.js'
import uuid from 'node-uuid'
import { fetchTodos, addTodo, toggleTodo, removeTodo } from '../actions'
import { getTodos } from '../selectors'

class Todos extends Component {
  componentDidMount () {
    this.fetchTodos()
  }

  fetchTodos = () => this.props.fetchTodos()

  addTodo = name => {
    const newTodo = { id: uuid.v4(), name, done: false }
    this.props.addTodo(newTodo)
  }

  removeTodo = id => {
    this.props.removeTodo(id)
  }

  toggleTodo = id => {
    this.props.toggleTodo(id)
  }

  render () {
    return (
      <div className='form-group'>
        <InputBar onSubmit={this.addTodo} />
        <ul className='list-group'>
          {this.props.todos.map(todo => (
            <Todo
              key={todo.id}
              todo={todo}
              onToggleTodo={this.toggleTodo}
              onRemoveTodo={this.removeTodo}
            />
          ))}
        </ul>
      </div>
    )
  }
}

// use reselect if getTodos is expensive
const mapStateToProps = state => ({
  todos: getTodos(state)
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      fetchTodos,
      addTodo,
      toggleTodo,
      removeTodo
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(Todos)
