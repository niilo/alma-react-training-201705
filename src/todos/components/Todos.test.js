import React from 'react'
import { shallow } from 'enzyme'
import { Todos } from './Todos'

const todo1 = [
  {
    id: '1',
    name: 'Go to gym',
    done: true
  }
]

const todo2 = [
  {
    id: '2',
    name: 'Learn some jest',
    done: false
  }
]

test('matches snapshot', () => {
  const todos = todo1.concat(todo2)
  expect(shallow(<Todos todos={todos} />)).toMatchSnapshot()
})
