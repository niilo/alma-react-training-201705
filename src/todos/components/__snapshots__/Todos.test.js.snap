exports[`test matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="form-group">
    <InputBar
        onSubmit={[Function]} />
    <ul
        className="list-group">
        <Todo
            onRemoveTodo={[Function]}
            onToggleTodo={[Function]}
            todo={
                Object {
                    "done": true,
                    "id": "1",
                    "name": "Go to gym",
                  }
            } />
        <Todo
            onRemoveTodo={[Function]}
            onToggleTodo={[Function]}
            todo={
                Object {
                    "done": false,
                    "id": "2",
                    "name": "Learn some jest",
                  }
            } />
    </ul>
</div>,
  "nodes": Array [
    <div
      className="form-group">
      <InputBar
            onSubmit={[Function]} />
      <ul
            className="list-group">
            <Todo
                  onRemoveTodo={[Function]}
                  onToggleTodo={[Function]}
                  todo={
                        Object {
                              "done": true,
                              "id": "1",
                              "name": "Go to gym",
                            }
                  } />
            <Todo
                  onRemoveTodo={[Function]}
                  onToggleTodo={[Function]}
                  todo={
                        Object {
                              "done": false,
                              "id": "2",
                              "name": "Learn some jest",
                            }
                  } />
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Todos
        todos={
                Array [
                        Object {
                          "done": true,
                          "id": "1",
                          "name": "Go to gym",
                        },
                        Object {
                          "done": false,
                          "id": "2",
                          "name": "Learn some jest",
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Todos {
        "_reactInternalInstance": [Circular],
        "addTodo": [Function],
        "context": Object {},
        "fetchTodos": [Function],
        "props": Object {
          "todos": Array [
            Object {
              "done": true,
              "id": "1",
              "name": "Go to gym",
            },
            Object {
              "done": false,
              "id": "2",
              "name": "Learn some jest",
            },
          ],
        },
        "refs": Object {},
        "removeTodo": [Function],
        "state": null,
        "toggleTodo": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="form-group">
          <InputBar
                    onSubmit={[Function]} />
          <ul
                    className="list-group">
                    <Todo
                              onRemoveTodo={[Function]}
                              onToggleTodo={[Function]}
                              todo={
                                        Object {
                                                  "done": true,
                                                  "id": "1",
                                                  "name": "Go to gym",
                                                }
                              } />
                    <Todo
                              onRemoveTodo={[Function]}
                              onToggleTodo={[Function]}
                              todo={
                                        Object {
                                                  "done": false,
                                                  "id": "2",
                                                  "name": "Learn some jest",
                                                }
                              } />
          </ul>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="form-group">
          <InputBar
                    onSubmit={[Function]} />
          <ul
                    className="list-group">
                    <Todo
                              onRemoveTodo={[Function]}
                              onToggleTodo={[Function]}
                              todo={
                                        Object {
                                                  "done": true,
                                                  "id": "1",
                                                  "name": "Go to gym",
                                                }
                              } />
                    <Todo
                              onRemoveTodo={[Function]}
                              onToggleTodo={[Function]}
                              todo={
                                        Object {
                                                  "done": false,
                                                  "id": "2",
                                                  "name": "Learn some jest",
                                                }
                              } />
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Todos
    todos={
        Array [
            Object {
              "done": true,
              "id": "1",
              "name": "Go to gym",
            },
            Object {
              "done": false,
              "id": "2",
              "name": "Learn some jest",
            },
          ]
    } />,
}
`;
